lib/route/nh.c

```diff
@@ -221,6 +221,20 @@ int rtnl_nh_get_oif(struct rtnl_nh *nexthop)
 	return 0;
 }
 
+int rtnl_nh_set_oif(struct rtnl_nh *nexthop, int ifindex)
+{
+	if (!nexthop)
+		return -NLE_INVAL;
+
+	if (ifindex <= 0)
+		return -NLE_RANGE;
```

Comment:
the default for the value is zero. So the setter should allow to reset the default.

It's only unclear, whether setting to 0 should also mean to unset NH_ATTR_OIF in the mask.

I think kernel would not differentiate between unset and (explictly set to) zero. Except for older kernels that didn't support NHA_OIF yet. But I think we should not take that as indication that libnl should differenciate. Meaning: explicitly setting to zero should be allowed, but it also means to clear NH_ATTR_OIF.

WDYT?